src/database/Database.cpp

```diff
@@ -335,6 +335,18 @@ Database::getUpsertTimer(std::string const& entityName)
         .TimeScope();
 }
 
+void
+Database::maybeArtificiallySleepForSimulatedLatency()
+{
+#ifdef BUILD_TESTS
+    auto usec = mApp.getConfig().ARTIFICIAL_DATABASE_LATENCY_FOR_TESTING;
+    if (usec.count() != 0)
+    {
+        std::this_thread::sleep_for(usec);
```

Comment:
Is this "good enough" (like did you actually look at something like Tracy plots for database access)?
If the goal is to simulate slow disks, the latency hit is on a per iops basis, not on a per query basis (and this depends on a lot of things).
For example, I would imagine that there is some function of log(table size):database cache size ratio. Also, read vs write performance could matter (especially when there are indexes that need to be updated as 1 sql op -> many iops).

Reply:
I have not yet examined tracy traces for comparison, will do so.

In terms of expectations: I believe there are two latency hits:
  - Round trip through socket to database process
  - Time to perform 1 fsync on database WAL

Granted the size of data to fsync will differ depending on the query, but I do not think it will differ by a lot -- I think the cost to push 1k and 1mb through to the disk is fairly similar (especially on spinning disk).

I believe a crude sleep like this therefore over-approximates the slowdown (as it penalizes reads as well as writes) but for purposes of simulation I think it's plausible.